### PR TITLE
Add track convenience method

### PR DESCRIPTION
**Pull request by RongxinZhang**
/cc @slajax @MadManMathew @RongxinZhang 

As discussed.

The track method is a convenience method that should prepend `track ` to the given type and assign the level `info`, and then call send.

- [x] Add a Logger#track(type, [msg]) method
- [x] Modify segment plugin to forward track messages
- [x] Update readme

### CODE REVIEW

**Comment by RongxinZhang** [comment]
@slajax @lvivier @MadManMathew plz review when you have time.

**Comment by MadManMathew** [comment]
pretty sure @lvivier was against making a new level.
I believe the track was suppose to mimic the info but with tracking functionality

File: README.md [hunk 1]
```diff
@@ -80,6 +80,54 @@ In general, the more serious the message, the "higher" the
...
+    - ```userId {string}```, something to identify the user by
+    - ```event {string}```, if empty will default to the ```type {string}```
+    - ```properties {object}```,anything other information that are valuable, and should contain a ```trait {object}``` for user details.
+
+#### an example
+
+```js
+// call with a descriptive context
+var log = require('log')('api:analytic')
+
```
Comment (lvivier) [inline README.md]: white space

File: README.md [hunk 2]
```diff
@@ -80,6 +80,54 @@ In general, the more serious the message, the "higher" the
...
+// 1. basic request with just userId. This will call Segment.Io's track() method
+log.track('api-request-listings', { userId: '550759f9a6fd33a61fffd70b' })    
+
+// 2. request with userId and events, the keen.io event will be api-request-agents. This will also call Segment.Io's track() method
+log.track('api-request-listings', { userId: '550759f9a6fd33a61fffd70b', event: 'api-request-agents'})    
+
+// 3. request with userId and events. This will also call Segment.Io's identify() method then track() method. 
+log.track('api-request-listings', { 
+                                    userId: '550759f9a6fd33a61fffd70b',
```
Comment (lvivier) [inline README.md]: can you collapse this white space, only need two spaces to indent

File: bin/logd [hunk 3]
```diff
@@ -28,6 +28,10 @@ var PORT = process.env.LOG_SERVER_PORT || 57000
 var HTTP_PORT = process.env.LOG_HTTP_PORT || 5000
 
 
+// TEMP
+var TEMP_LOGENTRIES_TOKEN = "01123fa6-e182-4528-ae53-e7821411305a";
```
Comment (lvivier) [inline bin/logd]: remove

File: bin/logd [hunk 4]
```diff
@@ -46,12 +50,10 @@ app
  * Notification plugin tokens
  * If tokens are in the environment, plugins will be loaded
  */
-
-if (!app.logentries) app.logentries = process.env.LOGENTRIES_APP_TOKEN
+if (!app.logentries) app.logentries = process.env.LOGENTRIES_APP_TOKEN || TEMP_LOGENTRIES_TOKEN
```
Comment (lvivier) [inline bin/logd]: ditto

File: lib/index.js [hunk 5]
```diff
@@ -89,14 +89,17 @@ Logger.prototype.send = function (level, type, msg) {
...
 
+
+// TODO: Set Default objects
   // msg obj
   var obj = {
     timestamp: Date.now(),
     message: msg,
-    level: level,
-    type: type,
+    level: level || "info",
```
Comment (lvivier) [inline lib/index.js]: don't mix single and double quotes

File: README.md (hunk 2 above)
Comment (RongxinZhang) [inline README.md]: cool. I'm doing this as part of https://github.com/Retsly/api/issues/1317

**Comment by RongxinZhang** [comment]
@MadManMathew I initially started with that, but I believe tracking is a separate action from info and don't think they should mix.

File: lib/index.js [hunk 6]
```diff
@@ -137,6 +140,31 @@ levels
     }
   })
 
+/**
+ * Logger track
+ *
+ * @param {String} userId - required - this is a keen event
```
Comment (lvivier) [inline lib/index.js]: This is [how you do parameters with properties in JSDoc](http://usejsdoc.org/tags-param.html#parameters-with-properties):

```js
/**
 * Logger track
 * @param {string} type
 * @param {object} msg
 * @param {string} msg.userId
 * @param {string} msg.event
 * @param {object} [msg.properties]
```

File: lib/index.js [hunk 7]
```diff
@@ -137,6 +140,31 @@ levels
...
+  if(!msg.event)
+    return msg.event = type //If event is empty use 'type'
+
+  this.send('track', type, msg)
+}
+
+// Error messages whem 'msg' object does not exist || does not have a userId
+function trackErrs(msg){
+  if(!msg)
+    throw new Error("Please add a bundle object");
```
Comment (lvivier) [inline lib/index.js]: Could use `TypeError` and using the same terminology ("msg") will help ppl out when they see this message

File: bin/logd (hunk 3 above)
Comment (RongxinZhang) [inline bin/logd]: Ok. 'm doing this as part of Retsly/api#1317

File: test/track.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+/**
```
Comment (lvivier) [inline test/track.js]: :beers: tests goood

Comment (RongxinZhang) [inline test/track.js]: yaaa!

**Comment by lvivier** [comment]
@RongxinZhang :+1: lookin good

**Comment by MadManMathew** [comment]
:+1:

**Pull request merged**